### PR TITLE
Add some feature to rgb/led matrix for wireless application

### PR DESCRIPTION
**Pull request by lokher**
## Description

Add LED/RGB_MATRIX_DRIVER_SHUTDOWN_ENABLE feature to save power.
Add LED/RGB_MATRIX_BRIGHTNESS_TURN_OFF_VAL feature to turn off backlight if brightness is too low.
Enhance LED_DISABLE_TIMEOUT feature.

## Types of Changes

- [x] Core
- [ ] Bugfix
- [x] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: quantum/led_matrix/led_matrix.c [hunk 1]
```diff
@@ -105,7 +105,11 @@ static effect_params_t led_effect_params = {0, LED_FLAG_ALL, false};
 static led_task_states led_task_state    = SYNCING;
 #if LED_DISABLE_TIMEOUT > 0
 static uint32_t led_anykey_timer;
+static uint32_t led_disable_timeout = LED_DISABLE_TIMEOUT;
```
Comment (drashna) [inline quantum/led_matrix/led_matrix.c:108]: `LED_DISABLE_TIMEOUT` has been renamed  to `LED_MATRIX_TIMEOUT`.  All instances of this needs to be updated.

File: quantum/rgb_matrix/rgb_matrix.c [hunk 2]
```diff
@@ -128,7 +128,11 @@ static effect_params_t rgb_effect_params = {0, LED_FLAG_ALL, false};
 static rgb_task_states rgb_task_state    = SYNCING;
 #if RGB_DISABLE_TIMEOUT > 0
 static uint32_t rgb_anykey_timer;
+static uint32_t rgb_disable_timeout = RGB_DISABLE_TIMEOUT;
```
Comment (drashna) [inline quantum/rgb_matrix/rgb_matrix.c:131]: Same here. `RGB_MATRIX_TIMEOUT`, in this case.

**Review by drashna — CHANGES_REQUESTED** [review]

File: quantum/led_matrix/led_matrix.c [hunk 3]
```diff
@@ -635,3 +669,31 @@ void led_matrix_set_flags(led_flags_t flags) {
...
+    led_disable_timeout = timeout;
+}
+void led_matrix_disable_time_reset(void) {
+    led_anykey_timer = 0;
+}
+#endif // LED_DISABLE_TIMEOUT > 0
+
+#ifdef LED_MATRIX_DRIVER_SHUTDOWN_ENABLE
+void led_matrix_driver_shutdown(void) {
+    led_matrix_driver.shutdown();
```
Comment (zvecr) [inline quantum/led_matrix/led_matrix.c:688]: This isnt safe, as the other backend drivers do not implement the function. They should be explicitly initialled to NULL and  `led_matrix_driver.shutdown`  checked before use

**Review by zvecr — CHANGES_REQUESTED** [review]
* `LED_MATRIX_DRIVER_SHUTDOWN_ENABLE` doeesnt really align with other configuration variables
* No docs have been updated

**Pull request closed without merging**